src/completions.ts

```diff
@@ -268,6 +272,10 @@ export abstract class CompletionSourceFuse extends CompletionSource {
         } else {
             this.deselect()
         }
+
+        // sort this.options by score
+        const sorted_options = matches.map(index => this.options[index])
+        this.options = sorted_options.concat(hidden_options)
```

Comment:
Need to check whether this overrides e.g. history sorting based on frequency.

Reply:
On an orthogonal topic, the history did not go through the fuse match as it override the `updateChain()` which goes through the default score related methods.